server/svix-server/src/queue/mod.rs

```diff
@@ -190,56 +218,80 @@ impl TaskQueueDelivery {
         }
     }
 
-    pub async fn ack(self) -> Result<()> {
+    pub async fn ack(mut self) -> Result<()> {
         tracing::trace!("ack {}", self.id);
-        run_with_retries(
-            || async {
-                match &self.acker {
-                    Acker::Memory(_) => Ok(()), // nothing to do
-                    Acker::Redis(q) => q.ack(&self.id, &self.task).await.trace(),
-                    Acker::RabbitMQ(delivery) => {
-                        delivery
-                            .ack(BasicAckOptions {
-                                multiple: false, // Only ack this message, not others
-                            })
-                            .await
-                            .map_err(Into::into)
+
+        let mut retry = Retry::new(should_retry, RETRY_SCHEDULE);
+        loop {
+            if let Some(result) = retry
+                .run(|| async {
+                    match &mut self.acker {
+                        Acker::Redis(q) => q.ack(&self.id, &self.task).await.trace(),
+                        Acker::RabbitMQ(delivery) => {
+                            delivery
+                                .ack(BasicAckOptions {
+                                    multiple: false, // Only ack this message, not others
+                                })
+                                .await
+                                .map_err(Into::into)
+                        }
+                        Acker::Omni(delivery) => delivery.ack().await.map_err(Into::into),
                     }
-                }
-            },
-            should_retry,
-            RETRY_SCHEDULE,
-        )
-        .await
+                })
+                .await
+            {
+                return result;
+            }
+        }
```

Comment:
This weird formulation of the retry loop is unfortunately required for the `&mut ack` design to pass the borrow checker. It's a bit hard to explain but I can attempt to write up the details if somebody is interested. Or we just change omniqueue's `ack` and `nack` methods to take `&self` ^^